### PR TITLE
Add split-layout content and example stubs

### PR DESCRIPTION
**Pull request by alkali47**

### CODE REVIEW

File: articles/ds/components/split-layout/index.asciidoc [hunk 1]
```diff
@@ -11,15 +11,156 @@ section-nav: incomplete
...
+include::{root}/frontend/demo/component/splitlayout/split-layout-initial-splitter-position.ts[render,tags=snippet,indent=0,group=TypeScript]
+----
+
+[source, java]
+----
+include::{root}/src/main/java/com/vaadin/demo/component/splitlayout/SplitLayoutInitialSplitterPosition.java[render,tags=snippet,indent=0,group=Java]
+----
+--
+
+The splitter will respect the minimum and maximum size of the content area components.
```
Comment (jouni) [inline articles/ds/components/split-layout/index.asciidoc:85]:
```suggestion
The splitter respects the minimum and maximum size of the content area components.
```

File: articles/ds/components/split-layout/index.asciidoc [hunk 2]
```diff
@@ -11,15 +11,156 @@ section-nav: incomplete
...
+----
+--
+
+== Orientation
+
+The default orientation is horizontal, placing the content areas side by side.
+The orientation can also be vertical.
+
+Orientation should be set based on the content and the screen size.
+Alternatively, allow the user to choose which orientation they wish to use.
```
Comment (jouni) [inline articles/ds/components/split-layout/index.asciidoc:48]:
```suggestion
The user can also be allowed to choose which orientation they wish to use.
```

File: articles/ds/components/split-layout/index.asciidoc [hunk 3]
```diff
@@ -11,15 +11,156 @@ section-nav: incomplete
...
+----
+
+[source, java]
+----
+include::{root}/src/main/java/com/vaadin/demo/component/splitlayout/SplitLayoutMinMaxSize.java[render,tags=snippet,indent=0,group=Java]
+----
+--
+
+The split can be adjusted programmatically, for example, by using a Button.
+This is useful when the user wants to quickly toggle between certain positions.
```
Comment (jouni) [inline articles/ds/components/split-layout/index.asciidoc:102]:
```suggestion
This is useful when the user wants to toggle between certain positions.
```

File: articles/ds/components/split-layout/index.asciidoc [hunk 4]
```diff
@@ -11,15 +11,156 @@ section-nav: incomplete
...
+----
+
+[source, java]
+----
+include::{root}/src/main/java/com/vaadin/demo/component/splitlayout/SplitLayoutThemeVariants.java[render,tags=snippet,indent=0,group=Java]
+----
+--
+
+The `small` theme variant makes the divider smaller.
+The `minimal` theme variant removes the divider completely.
```
Comment (jouni) [inline articles/ds/components/split-layout/index.asciidoc:137]:
```suggestion
The `minimal` theme variant hides the visual divider.
```

File: articles/ds/components/split-layout/index.asciidoc [hunk 5]
```diff
@@ -11,15 +11,156 @@ section-nav: incomplete
...
+Both variants only show the split handle on hover and are therefore not ideal for touch devices.
```
Comment (jouni) [inline articles/ds/components/split-layout/index.asciidoc:138]:
```suggestion
Both variants only show the split handle on hover and are not ideal for touch devices.
```

File: articles/ds/components/split-layout/index.asciidoc [hunk 6]
```diff
@@ -11,15 +11,156 @@ section-nav: incomplete
...
+
+[source, java]
+----
+include::{root}/src/main/java/com/vaadin/demo/component/splitlayout/SplitLayoutMinimalThemeVariants.java[render,tags=snippet,indent=0,group=Java]
+----
+--
+
+When using the `minimal` theme variant, it is recommended to somehow suggest the split between the two sides, for example by styling one side as a layer on top of the other.
+
+While these variants reduce the visual clutter, they make it less obvious to the user that the content is resizable.
```
Comment (jouni) [inline articles/ds/components/split-layout/index.asciidoc:156]: Add `resizable` to [.github/styles/Vocab/Docs/accept.txt](https://github.com/vaadin/docs/blob/latest/.github/styles/Vocab/Docs/accept.txt)

**Review by jouni — APPROVED** [review]
With the suggested fixes, 👍

File: articles/ds/components/split-layout/index.asciidoc (hunk 6 above)
Comment (Haprog) [reply articles/ds/components/split-layout/index.asciidoc:156]: Done.

**Comment by Haprog** [comment]
Rebased onto `latest`.

**Pull request merged**